pkg/virt-controller/watch/vmi/vmi.go

```diff
@@ -171,6 +172,10 @@ func (i informalSyncError) RequiresRequeue() bool {
 	return false
 }
 
+type statusUpdater interface {
```

Comment:
Since the updater object is an empty struct, how about simplifying this into:
```go
type statusUpdater func(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod) error
```

Reply:
Basically you are right, this decoupling could have been done using a function pointer since the concrete struct does not hold any state.

I personally think that having it as an interface provides a nicer semantic:
```go
err := c.networkStatusUpdater.Update(vmi, pod)
```

vs

```go
err := c.netStatusUpdater(vmi, pod)
```

It also allows us the flexibility to have state in the `netvmistatue.Updater` in case we will need it.

Another advantage, is the ability to have a struct as a stub.
If a function pointer is used, the stub will probably be a closure, which IMO requires a bit more mental energy to understand.

Reply:
That's fair, lgtm.